brd-ios/breadwallet/src/ViewModels/Transaction.swift

```diff
@@ -105,29 +112,50 @@ class Transaction {
         }
         return timestamp
     }
-
+    
     var hash: String { return transfer.hash?.description ?? "" }
-
+    
+    enum TransactionType {
+        case defaultTransaction
+        case swapTransaction
+    }
+    
+    var transactionType: TransactionType = .defaultTransaction
+    var swapTransationStatus: TransactionStatus?
+    var swapSource: Exchange.SourceDestination?
+    var swapDestination: Exchange.SourceDestination?
+    var swapOrderId: Int?
+    
     var status: TransactionStatus {
-        switch transfer.state {
-        case .created, .signed, .submitted, .pending:
-            return .pending
-        case .included:
-            
-            guard transfer.confirmation?.error == nil else {
+        switch transactionType {
```

Comment:
is this.. rly the best way? 🤔

Reply:
i don't see any other way tbh

Reply:
I don't see any other way. Since the old transactions have different logic inside the enum for pending and etc
![image](https://user-images.githubusercontent.com/99484553/181450684-ad3c1452-7c4b-475b-83ce-893b94ade361.png)